### PR TITLE
[node] Add structuredClone global to node

### PR DESCRIPTION
**Pull request by josh-**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nodejs/node/pull/39759 and [changelog item](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V17.md#2021-10-19-version-1700-current-bethgriggs)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by amcasey — COMMENTED** [review]
Please address the comment from @addaleax; otherwise, seems sensible to me.

File: types/node/globals.d.ts [hunk 1]
```diff
@@ -100,6 +100,13 @@ interface BigInt64Array extends RelativeIndexable<bigint> {}
 interface BigUint64Array extends RelativeIndexable<bigint> {}
 //#endregion ArrayLike.at() end
 
+/**
+ * @since 17.0.0
+ *
+ * Creates a deep clone of an object.
+ */
+declare function structuredClone<T>(value: T, transfer?: { transfer: any[] }): T;
```
[the remark this answers is not in the document]
Comment (josh-) [reply types/node/globals.d.ts]: Updated - thanks!

**Comment by josh-** [comment]
CI failures appear to be unrelated? Eg, running the `yargs` test for me locally passes:

```
$ npm test yargs
> definitely-typed@0.0.3 test
> dtslint types "yargs"

dtslint@0.0.111
testing from 3.9 to 4.1 in /Users/josh/Desktop/DefinitelyTyped/types/yargs/ts4.1
testing from 4.2 to 4.7 in /Users/josh/Desktop/DefinitelyTyped/types/yargs
```

**Review by addaleax — APPROVED** [review]

**Review by LinusU — APPROVED** [review]
Neat 👍

**Comment by amcasey** [comment]
Re-opening for fresh CI

**Comment by Araxeus** [comment]
Would love for this to get merged, this global is available in node since [v17.0.0](https://nodejs.org/en/blog/release/v17.0.0/) (6 months ago)

File: types/node/globals.d.ts [hunk 2]
```diff
@@ -100,6 +100,16 @@ interface BigInt64Array extends RelativeIndexable<bigint> {}
 interface BigUint64Array extends RelativeIndexable<bigint> {}
 //#endregion ArrayLike.at() end
 
+/**
+ * @since 17.0.0
```
Comment (amcasey) [inline types/node/globals.d.ts:104]: Nit: the other `@since` tags appear to use `v17.0.0`

**Review by amcasey — APPROVED** [review]
Sorry, I think the bot got confused and set the status incorrectly.

File: types/node/globals.d.ts (hunk 2 above)
Comment (josh-) [reply types/node/globals.d.ts:104]: Good pickup thanks, will update this instance and then fix [these other instances](https://cs.github.com/?scopeName=All+repos&scope=&q=repo%3ADefinitelyTyped%2FDefinitelyTyped+%2F%40since+%5Cd%2F+path%3A%2F%5Etypes%5C%2Fnode%5C%2F%2F) in a seperate PR 👍

**Review by amcasey — APPROVED** [review]

**Comment by josh-** [comment]
Ready to merge

**Pull request merged**